### DISCUSSION TITLE
Theming built-in components in v3?

### DISCUSSION THREAD

**Discussion by danielbuechele** [chakra-ui/chakra-ui#9242] [Q&A]
In v2 I used to change component styles in my theme like this:
```
  components: {
    Input: {
      baseStyle: {
        bg: 'blue.100',
      },
    }
```

What is the recommended way to style built in components in v3? I could create my own component, which in this case would render `<Input bg="blue.100" />`, but then I need to make sure everyone in my project is using my custom component rather than the built in Input exported from chakra-ui. Is there a way to style the built-in components like in v2?

**Comment by FloChehab** [comment]
I have spent quite some time in the V3 doc, and I don't understand too what is the recommended way of styling (snipet) components in the V3.

For instance if I want to style all dialog backdrops across an app, how should we do that ?

**Comment by karolinelemos** [comment]
I don't know if this is the best way, but I customized it using [recipes](https://www.chakra-ui.com/docs/theming/recipes): 

theme.js
```
const theme = createSystem(defaultConfig, {
  theme: {
    recipes: {
      input: inputRecipe,
    },
  }
})
```
inputRecipe.js
```
export const inputRecipe = defineRecipe({
  base: {
      width: '100%',
  },
  variants: {
    visual: {
        custom: {
            bg: 'blue.100'
        },
    }
 }
})
```

`<Input visual='custom' />`

**Reply by karolinelemos** [reply]
It's possible set the default to apply to all Inputs.

```
defaultVariants: {
   visual: "custom",
}
```

**Comment by FloChehab** [comment]
Thanks @karolinelemos. I am trying to do something similar for the Card component, but it doesn't seem to work.

@segunadebayo I am sorry to ping you here, but I feel this is quite core to the new v3 and I still can't figure out how to do such things. Thanks in advance for your help.

**Reply by karolinelemos** [reply]
@FloChehab For the card, use [slot recipes](https://www.chakra-ui.com/docs/theming/slot-recipes)
Like this: 

```
theme: {
  slotRecipes: {
      card: defineSlotRecipe({
        variants: {
          visual: { // to custom your visual
            'my-visual': {
              root: {
                bg: 'red',
              },
            }
          },
          variant: {
            outline: { // to override the style of variant
              root: {
                bg: 'blue',
              },
            },
          },
        },
      }),
    },
}
```

```
<Card.Root visual="my-visual">
  <Card.Header>test</Card.Header>
</Card.Root>
```
or
```
<Card.Root variant="outline">
  <Card.Header>test</Card.Header>
</Card.Root>
```

**Reply by OmarDronodat** [reply]
one last question for stack like hstack and vstack how we are supposed to do it i am trying to do it like the input but it seems not working

**Reply by OmarDronodat** [reply]
and is there any forward documentation how can i apply the recipes to different components in our themes as its only written for buttons
